core/positions/market_position.go

```diff
@@ -230,3 +230,20 @@ func (p MarketPosition) OrderReducesExposure(ord *types.Order) bool {
 	}
 	return false
 }
+
+// OrderReducesOnlyExposure returns true if the order reduce the position and the extra size if it was to flip the position side.
+func (p MarketPosition) OrderReducesOnlyExposure(ord *types.Order) (reduce bool, extraSize uint64) {
```

Comment:
I think this would make more sense as:

```
func (p MarketPosition) GetOrderExposureReducingSize(ord *types.Order) uint64 {
     if p.Size() == 0 || (p.Size() < 0 && ord.Side == types.SideSell) || (p.Size() > 0 && ord.Side == types.SideBuy) {
             return 0
     }
     size := p.Size()
     if size < 0 {
        size = -size
     }
    extra := int64(ord.Remaining)-size
    if extra < 0 {
        return 0
    }
    return extra
}
     
```

Reply:
hum correct. my brain is broken.

Reply:
OK, we seem to implement this logic in different places. It used to sit in execution package, I believe @witgaw moved it to risk (not sure exactly where, though). Maybe we should move it to a receiver on the order type, and pass it a position as an argument

Reply:
hum. they are doing slighlty different things to be fair.